zerver/tornado/event_queue.py

```diff
@@ -1556,6 +1570,17 @@ def reformat_legacy_send_message_event(
     return (modern_event, user_dicts)
 
 
+def process_user_group_name_update_event(event: Mapping[str, Any], users: Iterable[int]) -> None:
+    user_group_event = dict(event)
+    event_for_deactivated_group = user_group_event["data"].pop("deactivated", False)
```

Comment:
This detail deserves a comment, both here and in `actions/user_groups.py`, that this field is popped here and not part of the API.

Might be better to have it not be in the `data` at all, but instead at the root of the event object? Not sure, but that seems like it might be cleaner.